tendrl/commons/message.py

```diff
@@ -50,6 +50,8 @@ def __init__(self, priority, publisher, payload, job_id=None,
         else:
             self.caller = caller
         self.priority = priority
+        if not publisher:
+            publisher = NS.publisher_id
```

Comment:
I guess the issue would be when publisher_id itself is not present in NS, it will throw a KeyError which is not handled here, I suppose if what you suggested should be implemented then either we will have to handle this error case or make sure no one calls this function until NS.publisher_id is atleast present i.e NS has publisher _id key in its dict